### DISCUSSION TITLE
O365 auth Caught exception while requesting OAuth 2.0

### DISCUSSION THREAD

**Discussion by davidstanaway** [simonrob/email-oauth2-proxy#312] [General]
Caught exception while requesting OAuth 2.0 credentials for account my@address.com: URLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1123)'))

[my@address.com]
permission_url = https://login.microsoftonline.com/mytenantid/oauth2/v2.0/authorize
token_url = https://login.microsoftonline.com/mytenantid/oauth2/v2.0/token
oauth2_scope = https://outlook.office.com/IMAP.AccessAsUser.All https://outlook.office.com/POP.AccessAsUser.All https://outlook.office.com/SMTP.Send offline_access
redirect_uri = http://localhost
client_id = myappid

[emailproxy]
use_login_password_as_client_credentials_secret = True

Email client is thunderbird with IMAP (Will be using same config for someone that needs to use Eudora [Don't ask!] once validated and I am happy with the config.

App registration has these permissions
Microsoft Graph 
IMAP.AccessAsUser.All | Delegated | Read and write access to mailboxes via IMAP.
POP.AccessAsUser.All | Delegated | Read and write access to mailboxes via POP.
SMTP.Send | Delegated | Send emails from mailboxes using SMTP AUTH.
User.Read

Log context: Authentication is successful, first use has be approve the application which is done.

2024-12-14 00:08:16: Authorisation request received for my@address.com (interactive mode)
2024-12-14 00:08:16: Please authorise your account my@address.com from the menu
2024-12-14 00:08:16: Authorisation request received for my@address.com (interactive mode)
2024-12-14 00:08:16: Please authorise your account my@address.com from the menu
2024-12-14 00:09:43: Returning authorisation request result for my@address.com
2024-12-14 00:09:43: Returning authorisation request result for my@address.com
2024-12-14 00:09:43: Authentication completed for my@address.com
2024-12-14 00:09:43: Caught exception while requesting OAuth 2.0 credentials for account my@address.com: URLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1123)'))
2024-12-14 00:09:43: Caught exception while requesting OAuth 2.0 credentials for account my@address.com: URLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1123)'))

I am not sure what certificate is expired. the tenant/directory ID is correct and the login URLs are successfully accessed in my browser without cert issues.

**Comment by cmchick** [comment]
Expired certs might mean your date and time is not set correctly. Verify that your host's clock is accurate.

Coincidentally I am doing the same thing: testing with Thunderbird with intent to deploy for a coworker who is metaphorically married to Eudora.

**Reply by Al-clueless** [reply]
Hi, 
like your co-worker, I LOVE Eudora.  It has it's problems, like this one, but for me it works better than anything else I have tried.  I am muddling through with gmail for the past month and I really don't like it.  I'm making small baby steps toward getting Eudora to work again, but if you could share how you got your co-workers Eudora to work, it would be greatly appreciated.  TIA

**Reply by GAllen27** [reply]
Eudora Qs are best posted to the Eudora list.  http://listmoms.net/eudora-win/index.html

A donation of any amount will get you access to Pete Maclean's update of the Eudora source code:
https://www.indiegogo.com/projects/hermes-eudoramail-8-0-the-final-stretch
This is the future of Eudora. But the host of this igg page is sometimes absent and it may take a while to get access to the alpha bits and to the Signal group where more detailed discussions are.

**Reply by simonrob** [reply]
It's also worth searching the previous issues and discussions in this repository for Eudora-related solutions. In general, though, I can't really offer specific advice for a client that I've never used myself. You might have more luck with the [proxy's sponsor](https://auth-email.com/) as they provide a custom service.

**Comment by simonrob** [comment]
In addition to the date/time suggestion (which is valid), I'd also recommend taking a look at previous similar issues (e.g., [this one](https://github.com/simonrob/email-oauth2-proxy/issues/259)). You might benefit from installing [`certifi`](https://github.com/certifi) (`python -m pip install certifi`) – even if the browser certificates are up-to-date, the ones used by the proxy may not be.
